form/src/components/Form.js

```diff
@@ -0,0 +1,47 @@
+import React, { useState } from "react";
+import Question from "./Question";
+import Result from "./Result";
+
+const Form = () => {
+  const [answer, setAnswer] = useState(" ");
+  const [isSubmit, setIsSubmit] = useState(false);
+  const handleCheckInput = (e) => {
+    setAnswer(e.target.value);
+  };
+  const handleClick = (e) => {
+    e.preventDefault();
+    if (
+      answer.toLowerCase().trim() === "yes" ||
+      answer.toLowerCase().trim() === "no"
+    ) {
+      alert(`question : ${answer.trim()} submitted`);
+      setIsSubmit(true);
+    } else {
+      alert("you just passed wrong asnwer you should use yes or no as answer");
+    }
+  };
+
+  return (
+    <div>
+      <form>
+        <label htmlFor="answer">
+          <Question />
+          <input
+            id="answer"
+            type="text"
+            value={answer}
+            required
+            onChange={handleCheckInput}
+            placeholder="yes or no"
+          />
+        </label>
+        <button type="submit" onClick={handleClick}>
```

Comment:
connect event handler to from's `onSubmit` attribute: https://www.w3schools.com/jsref/event_onsubmit.asp, not to the button.